wave_lang/kernel/wave/templates/speculative_decoding.py

```diff
@@ -543,12 +435,56 @@ def else_():
             outer_done,
         ) = outer_loop
 
-        tkw.write(
-            last_accepted_retrieve_idx,
-            last_accepted_retrieve_idx_vec,
-            elements_per_thread=1,
-        )
-        tkw.write(cur_prob_offset, cur_prob_offset_vec, elements_per_thread=1)
         tkw.write(num_accepted_tokens, accept_token_num, elements_per_thread=1)
```

Comment:
Do we need to write num_accepted_tokens ?

Reply:
There is an assertion in the test script. It is mutable, hence we need to write it.